File: src/main/java/cc/suvankar/moneytrail/contact/dto/ContactResponse.java
package cc.suvankar.moneytrail.contact.dto;

import cc.suvankar.moneytrail.contact.Contact;
import java.time.OffsetDateTime;
import java.util.Objects;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ContactResponse {
  private UUID contactId;
  private String name;
  private String email;
  private String phoneNo;
  private String notes;
  private OffsetDateTime createdAt;
  private OffsetDateTime updatedAt;

  public static ContactResponse from(Contact contact) {
    return new ContactResponse(
        Objects.requireNonNull(contact.getId(), "Contact has no id"),
        contact.getName(),
        contact.getEmail(),
        contact.getPhoneNo(),
        contact.getNotes(),
        Objects.requireNonNull(contact.getCreatedAt(), "Contact has no createdAt"),
        Objects.requireNonNull(contact.getUpdatedAt(), "Contact has no updatedAt"));
  }
}

File: src/main/kotlin/cc/suvankar/moneytrail/contact/dto/ContactResponse.kt
package cc.suvankar.moneytrail.contact.dto

import cc.suvankar.moneytrail.contact.Contact
import java.time.OffsetDateTime
import java.util.UUID

data class ContactResponse(
  val id: UUID,
  val name: String,
  val email: String?,
  val phoneNo: String?,
  val notes: String?,
  val createdAt: OffsetDateTime,
  val updatedAt: OffsetDateTime,
) {
  companion object {
    @JvmStatic
    fun from(contact: Contact) =
      ContactResponse(
        id = contact.id ?: error("Contact has no id"),
        name = contact.name,
        email = contact.email,
        phoneNo = contact.phoneNo,
        notes = contact.notes,
        createdAt = contact.createdAt ?: error("Contact has no createdAt"),
        updatedAt = contact.updatedAt ?: error("Contact has no updatedAt"),
      )
  }
}
